### DISCUSSION TITLE
Should glz::json_t be renamed to glz::json?

### DISCUSSION THREAD

**Discussion by stephenberry** [stephenberry/glaze#1313] [Ideas]
I'm leaning towards renaming `glz::json_t` to `glz::json`. This makes `glz::json` naming match better with other DOM libraries.

This is a major breaking change, but it isn't difficult to refactor.

**Comment by stephenberry** [comment]
I think in the long term `glz::json_t` will be renamed `glz::json`. We can first change the type name, alias to `glz::json_t` and mark it with [[deprecated]].

Additionally, `glz::json_t` is useful beyond JSON for generic handling. The BEVE format in Glaze supports it. So, I think an alias called `glz::generic` or `glz::dom` would also be useful, not to replace `glz::json`, but to provide another way to name generic handling.

**Comment by StormLord07** [comment]
I think that's going to be pretty useful, even just for clarity. If I remember correctly, the `_t` postfix is usually reserved for typedefs, while `glz::json_t` is a struct itself, not a typedef.

I haven’t really used BEVE with `glz::json_t`, though I did manage to get gRPC working with structs and BEVE.

Wouldn’t it make more sense to keep `json_t` as a typedef and make `glz::generic` (or similar) for cases where it’s meant to be used across JSON, BEVE, or other formats? That way the naming is clearer, and we keep a distinction between the specific JSON and a more general-purpose type? That way the change isnt as breaking.

**Comment by stephenberry** [comment]
In progress here: #1984
